File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/payment/TransactionId.java
package it.giannibombelli.wsc2026.payment.domain.payment;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.domain.identity.EntityId;

import java.util.Objects;
import java.util.UUID;

public record TransactionId(UUID value) implements EntityId {
    public TransactionId {
        Require.requireArgument(value, "value");
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        return Objects.equals(value, ((TransactionId) o).value);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(value);
    }

    @Override
    public String toString() {
        return value.toString();
    }
}

File: typescript/src/payment/domain/payment/transactionId.ts
// Identità di una Transaction all'interno di un Payment.

import { requireArgument } from '@/common/utils/requireArgument.js';
import type { EntityId } from '@/common/domain/identity/entityId.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';

export class TransactionId implements EntityId {
  readonly value: Uuid;

  constructor(value: Uuid) {
    requireArgument(value, 'TransactionId value');
    this.value = value;
  }

  equals(other: TransactionId): boolean {
    return this.value.value === other.value.value;
  }

  hashCode(): number {
    let hash = 0;
    const value = this.value.value;
    for (let i = 0; i < value.length; i++) {
      hash = (hash << 5) - hash + value.charCodeAt(i);
      hash |= 0;
    }
    return hash;
  }
}
